src/main/java/seedu/exercise/model/ScheduleBook.java

```diff
@@ -0,0 +1,104 @@
+package seedu.exercise.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.exercise.model.schedule.Schedule;
+import seedu.exercise.model.schedule.UniqueScheduleList;
+
+/**
+ * Wraps all data at the schedule-book level.
+ * Conflicting schedules are not allowed.
+ */
+public class ScheduleBook implements ReadOnlyScheduleBook {
+
+    private final UniqueScheduleList schedules;
+
+    {
+        schedules = new UniqueScheduleList();
+    }
+
+    public ScheduleBook() {
+
+    }
+
+    public ScheduleBook(ReadOnlyScheduleBook toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    /**
+     * Resets the existing data of this {@code ScheduleBook} with {@code newData}.
+     */
+    public void resetData(ReadOnlyScheduleBook newData) {
+        requireNonNull(newData);
+
+        setRegimes(newData.getScheduleList());
+    }
+
+    /**
+     * Replaces the contents of the regime list with {@code regimes}.
+     * {@code regimes} must not contain duplicate regimes.
+     */
+    public void setRegimes(List<Schedule> schedules) {
+        this.schedules.setSchedules(schedules);
+    }
+
+
+    /**
+     * Returns true if a schedules with the same date exists in the schedules book.
```

Comment:
Should be "if a schedule with the same date..."
No plural in this case. 
Similar issues in the next few comments too.

Reply:
👍